src/fmt_md.rs

```diff
@@ -85,26 +84,6 @@ where
     writer_state.write_definitions(out, DefinitionsToWrite::Both, nodes_count > 1);
 }
 
-pub fn write_md_inlines<'a, I, W>(out: &mut Output<W>, nodes: I, inlines_writer: &mut MdInlinesWriter<'a>)
-where
-    I: Iterator<Item = MdElemRef<'a>>,
-    W: SimpleWrite,
-{
-    let mut writer_state = MdWriterState {
-        opts: &MdOptions {
-            link_reference_placement: ReferencePlacement::Doc, // but we won't actually write them
-            footnote_reference_placement: ReferencePlacement::Doc, // ditto
-            inline_options: MdInlinesWriterOptions {
-                link_format: LinkTransform::Keep, // unused here, but removing it is more refactoring than it's worth
```

Comment:
The fact that I had all these unused options should have keyed me into the fact that something weird was going on! The `link_format` one I had even noted as a TODO to clean up earlier, but I decided it wasn't worth it in 329d1e4c3db91fdd8fddebcc52132b2bbf491b99. I guess now it's done right!